vm-virtio/src/block.rs

```diff
@@ -530,7 +530,7 @@ impl<T: DiskFile> Drop for Block<T> {
 }
 
 impl<T: 'static + DiskFile + Send> VirtioDevice for Block<T> {
-    fn device_type(&self) -> u32 {
+    fn device_type_num(&self) -> u32 {
```

Comment:
Why did you modify this function name?

Reply:
@sboeuf @sameo 
> * as we don't want to loose the msix capability for our virtio devices, we need a more generic way to handle interrupts from the vm-device itself.

The most key part is ioapic and msi callback.  In vm-device usage, we need `Device::set_resources()` to set some resources back. But ioapic is not part of resources that allocated from vm-allocator, and also no special resource to set (I mean we just want to give cb back, or userspace_ioapic parameter instead, but it's not a kind of resource or high level things).
We might need think about:
how to set userspace_ioapic parameter to device, in order to create callback inside device.